client/src/app/pages/applications/components/application-assessment-status/application-assessment-status.tsx

```diff
@@ -15,77 +16,11 @@ export const ApplicationAssessmentStatus: React.FC<
 > = ({ application, assessments, archetypes, isLoading }) => {
   const { t } = useTranslation();
 
-  const applicationAssessments = assessments?.filter(
-    (assessment: Assessment) => assessment.application?.id === application.id
+  const assessmentStatusInfo = useAssessmentStatus(
+    assessments,
+    archetypes,
```

Comment:
I wonder if it would be easier just to have `useAssessmentStatus` just pull the assessments and archetypes directly since I don't think they're being further processed beyond what their fetch hooks are doing.

In general I'm thinking a custom data hook approach of only providing the differentiating information to really focus on what the component cares about will be helpful.  In this case the component only cares about the application it needs to render something for.

Reply:
Yeah good call. I made those changes.